src/planscape/forsys/get_forsys_inputs.py

```diff
@@ -0,0 +1,123 @@
+    save_debug_info: bool
+    # The planning region.
+    region: str
+    # Conditions whose AP scores will be considered when ranking projects.
+    priorities: list[str]
+    # Project areas to be ranked. A project area may be a multipolygon.
+    project_areas: dict[int, MultiPolygon]
+
+    def __init__(self, params: QueryDict) -> None:
+        if bool(params.get(self._URL_USE_ONLY_URL_PARAMS, False)):
+            # This is used for debugging purposes.
+            self._read_url_params_with_defaults(params)
+            self.save_debug_info = True
+        else:
+            self._read_db_params(params)
+            self.save_debug_info = False
+
+    def _read_url_params_with_defaults(self, params: QueryDict) -> None:
+        self.region = params.get(
+            self._URL_REGION, self._DEFAULT_REGION)
+        self.priorities = params.getlist(
+            self._URL_PRIORITIES, self._DEFAULT_PRIORITIES)
+        if self._URL_PROJECT_AREAS in params:
+            self.project_areas = self._read_project_areas_from_url_params(
+                params)
+        else:
+            self.project_areas = self._get_default_project_areas()
+
+    def _read_db_params(self, params: QueryDict) -> None:
+        # TODO: add db read logic.
+        raise Exception(
+            'WIP. ' +
+            'Please set set_all_params_via_url_with_default_values to true.')
+
+    def _get_default_project_areas(self) -> dict[int, MultiPolygon]:
+        srid = 4269
+        p1 = Polygon(((-120.14015536869722, 39.05413814388948),
+                     (-120.18409937110482, 39.48622140686506),
+                     (-119.93422142411087, 39.48622140686506),
+                     (-119.93422142411087, 39.05413814388948),
+                     (-120.14015536869722, 39.05413814388948)))
+        p1.srid = srid
+        p2 = Polygon(((-120.14015536869722, 38.05413814388948),
+                     (-120.18409937110482, 38.48622140686506),
+                     (-119.93422142411087, 38.48622140686506),
+                     (-119.93422142411087, 38.05413814388948),
+                     (-120.14015536869722, 38.05413814388948)))
+        p2.srid = srid
+        p3 = Polygon(((-121.14015536869722, 39.05413814388948),
+                     (-121.18409937110482, 39.48622140686506),
+                     (-120.53422142411087, 39.48622140686506),
+                     (-120.53422142411087, 39.05413814388948),
+                     (-121.14015536869722, 39.05413814388948)))
+        p3.srid = srid
+        m1 = MultiPolygon(p1, p2)
+        m1.srid = srid
+        m2 = MultiPolygon(p3)
+        m2.srid = srid
+        return {1: m1,
+                2: m2}
+
+    def _read_project_areas_from_url_params(
+            self, params: QueryDict) -> dict[int, MultiPolygon]:
+        project_areas = {}
+        for project_area_str in params.getlist(self._URL_PROJECT_AREAS):
+            project_area = ProjectAreaFromUrlParams(
```

Comment:
should these be already formatted as multipolygons by the time we reach this step?

Reply:
ProjectAreaFromUrlParams is a TypedDict object - not quite a multipolygon yet.
Below, I convert polygon coordinates into polygons, and then join them to create a multipolygon that's returned in the output dict.